tests/test_command_line.py

```diff
@@ -1156,6 +1156,41 @@ def test_generate_c_source_database_name(self):
                 self.assert_files_equal(database_c,
                                         'tests/files/c_source/' + database_c)
 
+    def test_generate_c_source_output_directory(self):
+        databases = [
+            'motohawk',
+        ]
+        
+        output_directory = 'some_dir'
+
+        for database in databases:
+            argv = [
+                'cantools',
+                'generate_c_source',
+                '--output-directory', output_directory,
+                'tests/files/dbc/{}.dbc'.format(database)
+            ]
+
+            database_h = os.path.join(output_directory, f'{database}.h')
+            database_c = os.path.join(output_directory, f'{database}.c')
+
+            if os.path.exists(database_h):
+                os.remove(database_h)
+
+            if os.path.exists(database_c):
+                os.remove(database_c)
+            
+            if os.path.exists(output_directory):
+                os.rmdir(output_directory)
```

Comment:
I suggest replacing all three removes with a single `shutil.rmtree(output_directory, ...)`.

Reply:
The actual solution is to use temporary directories everywhere as then you never have to think about removing files to begin with nor if where you are working is clean.

Reply:
I want to keep the output to manually look at the generated files sometimes. But I agree, all created folders should not be created in the root, but somewhere else. That's for later, if ever.